### DISCUSSION TITLE
Migrate kubescape package

### DISCUSSION THREAD

**Discussion by HollowMan6** [chocolatey-community/chocolatey-packages#2186] [General]
I would like to request migrating the [kubescape](https://community.chocolatey.org/packages/kubescape) package to here.

- Why it should be migrated to this repository: As [kubescape upstream](https://github.com/kubescape/kubescape) have pre-built binaries in their release page and they release new versions frequently.
- A link to both the package and the repository the source code is located in (if it exists): 
  - https://community.chocolatey.org/packages/kubescape
  - https://github.com/HollowMan6/chocolatey-packages/tree/master/automatic/kubescape (It's ready for a PR at branch [submit](https://github.com/HollowMan6/chocolatey-packages/tree/submit/automatic/kubescape))

- Who will be responsible for fixing issues, reviewing changes, and making sure it is kept up to date (ie, fix the updater when it fails): Me as well as members from the kubescape org

**Comment by AdmiringWorm** [comment]
> Why it should be migrated to this repository: As [kubescape upstream](https://github.com/kubescape/kubescape) have pre-built binaries in their release page and they release new versions frequently.

The reason specified doesn't sound like a reason to migrate the package to this repository, and it needs to answer the question correctly, so let me ask differently.

What would be the benefits of migrating this package to this repository instead of keeping it in your own repository?

**Comment by HollowMan6** [comment]
> What would be the benefits of migrating this package to this repository instead of keeping it in your own repository?

Hi, Thank you for your response! Kubescape is a [CNCF sandbox project](https://landscape.cncf.io/card-mode?project=sandbox&selected=kubescape), it is an important package for the community as a whole. In addition, I'm actually only a student doing a short term intern at CNCF: https://github.com/kubescape/kubescape/issues/400#issue-1144958944 Although later I can help fixing issues, reviewing changes, and making sure it is kept up to date for this package after this internship, it would be better to keep it to this repository where it can be maintained by a team who can keep it up-to-date in a longer term.

**Reply by HollowMan6** [reply]
@AdmiringWorm

**Reply by AdmiringWorm** [reply]
I need to talk with the rest of the team maintainers about whether we want to accept new package migrations (We are currently updating our Contribution Guidelines). I hope to bring this up on Monday. Otherwise, it will happen on the 5th or 6th of April before I can answer whether we will accept this migration.

**Reply by AdmiringWorm** [reply]
We've discussed accepting kubescape as a migrated package into this repository and agreed to accept it.

However, as I already mentioned, we are in the process of rewriting the Contribution Guidelines. As such, we ask you to make sure of one of the following:

- Wait until PR #2155 is merged before you create a pull request (this is the first iteration of the new guidelines)
- When creating the PR, make sure the package, your commits, and PR details are up to date with the new guidelines in the mentioned PR (do expect changes to the Guidelines as well that may be asked of you).

Also, we briefly looked at the package you pushed, and I'll make sure to mention at least one change that we would ask of you (so we don't need to ask for it during the PR review).
Currently, you are embedding an extensionless binary file called `kubescape-windows-latest`. Before embedding the file, this should be renamed to `kubescape.exe` instead of doing it as part of the installation. This way, the `chocolateyInstall.ps1` script can be removed as well (how to use the program or link to its usage should be in the package description, not in the output from the installation).

**Reply by AdmiringWorm** [reply]
I have created issue #2190, which the PR created needs to target. If you could comment on that issue, I can assign it to you.

**Reply by HollowMan6** [reply]
> We've discussed accepting kubescape as a migrated package into this repository and agreed to accept it.
> 
> However, as I already mentioned, we are in the process of rewriting the Contribution Guidelines. As such, we ask you to make sure of one of the following:
> 
> * Wait until PR [(#1754) Update documentation about contributing to repo #2155](https://github.com/chocolatey-community/chocolatey-packages/pull/2155) is merged before you create a pull request (this is the first iteration of the new guidelines)
> * When creating the PR, make sure the package, your commits, and PR details are up to date with the new guidelines in the mentioned PR (do expect changes to the Guidelines as well that may be asked of you).
> 
> Also, we briefly looked at the package you pushed, and I'll make sure to mention at least one change that we would ask of you (so we don't need to ask for it during the PR review). Currently, you are embedding an extensionless binary file called `kubescape-windows-latest`. Before embedding the file, this should be renamed to `kubescape.exe` instead of doing it as part of the installation. This way, the `chocolateyInstall.ps1` script can be removed as well (how to use the program or link to its usage should be in the package description, not in the output from the installation).

Okay, Thank you! I will discuss with the kubescape community about the change request.

**Reply by HollowMan6** [reply]
Hi @AdmiringWorm ! I have introduced the `kubescape.exe` into the Release Assets in v2.3.0, so I think the package is now ready for the PR. https://github.com/HollowMan6/chocolatey-packages/tree/submit

However, I can see that the PR you mentioned for updating the Contribution Guidelines is still unmerged. So I'm checking if you have any rough idea for the plans of when it can get merged. Thank you!

> However, as I already mentioned, we are in the process of rewriting the Contribution Guidelines. As such, we ask you to make sure of one of the following:
> 
> * Wait until PR [(#1754, #1940) Update documentation about contributing to repo #2155](https://github.com/chocolatey-community/chocolatey-packages/pull/2155) is merged before you create a pull request (this is the first iteration of the new guidelines)
> * When creating the PR, make sure the package, your commits, and PR details are up to date with the new guidelines in the mentioned PR (do expect changes to the Guidelines as well that may be asked of you).
> 
> Also, we briefly looked at the package you pushed, and I'll make sure to mention at least one change that we would ask of you (so we don't need to ask for it during the PR review). Currently, you are embedding an extensionless binary file called `kubescape-windows-latest`. Before embedding the file, this should be renamed to `kubescape.exe` instead of doing it as part of the installation. This way, the `chocolateyInstall.ps1` script can be removed as well (how to use the program or link to its usage should be in the package description, not in the output from the installation).

**Reply by AdmiringWorm** [reply]
@HollowMan6 Unfortunately, I don't have any eta on when it will be merged.

I have a few items to fix up after the last time it was reviewed before it can be reviewed again and hopefully merged.

Assuming your package is up to date with the guidelines in that PR, then feel free to open up a PR, which can then be reviewed.

**Reply by HollowMan6** [reply]
PR opened at https://github.com/chocolatey-community/chocolatey-packages/pull/2226
